client/src/hooks/http_hook.js

```diff
@@ -0,0 +1,40 @@
+import {useCallback, useState} from 'react';
+import axios from 'axios';
+
+const baseUrl = 'http://localhost:4000/api/';
+
+const base_axios = axios.create({
+  baseURL: baseUrl
+});
+
+export const useHttp = () => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const get_request = useCallback(async (url) => {
```

Comment:
Please, use camelCase in functions and instances.
https://github.com/airbnb/javascript#naming--camelCase